examples/twittermap/web/app/actor/TwitterMapPigeon.scala

```diff
@@ -0,0 +1,43 @@
+package actor
+
+import java.net.URI
+
+import akka.actor._
+import akka.stream.Materializer
+import org.eclipse.jetty.websocket.client.{ClientUpgradeRequest, WebSocketClient}
+import play.api.libs.json.JsValue
+import socket.TwitterMapClientSocket
+
+import scala.concurrent.ExecutionContext
+
+class TwitterMapPigeon (val cloudberryWS: String,
+                        val out: ActorRef)
+                       (implicit ec: ExecutionContext, implicit val materializer: Materializer) extends Actor with ActorLogging {
+
+  val twitterMapWebSocketClient: WebSocketClient = new WebSocketClient
+  val twitterMapClientSocket: TwitterMapClientSocket = new TwitterMapClientSocket(out)
+
+  override def preStart(): Unit = {
+    super.preStart
+    twitterMapWebSocketClient.start()
+    twitterMapWebSocketClient.connect(twitterMapClientSocket, new URI(cloudberryWS), new ClientUpgradeRequest)
+    twitterMapClientSocket.getLatch.await()
```

Comment:
what's the purpose of `getLatch` and `await`?

Reply:
The `countDownLatch` is used for concurrent control which guaranteed that the socket session is not accessed, interrupted or closed by any other thread. The connection is open until we call `close` method when the actor is killed. I saw many code samples are using it to secure the websocket connection and I thought we need it as well in our case so I put it here. It is open to discussion.

Some source here:
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CountDownLatch.html
https://github.com/jetty-project/embedded-jetty-websocket-examples/pull/2/files